gtsam/slam/dataset.h

```diff
@@ -243,6 +246,72 @@ struct SfmTrack {
   void add_measurement(size_t idx, const gtsam::Point2& m) {
     measurements.emplace_back(idx, m);
   }
+
+  /** Serialization function */
+  friend class boost::serialization::access;
+  template<class ARCHIVE>
+  void serialize(ARCHIVE & ar, const unsigned int /*version*/) {
+    ar & p;
+    ar & r;
+    ar & g;
+    ar & b;
+    ar & measurements;
+    ar & siftIndices;
+  }
+
+  /// assert equality up to a tolerance
+  bool equals(const SfmTrack &sfmTrack, double tol = 1e-9) const {
+    // check the 3D point
+    if (!p.isApprox(sfmTrack.p)) {
+      return false;
+    }
+
+    // check the RGB values
+    if (r!=sfmTrack.r || g!=sfmTrack.g || b!=sfmTrack.b) {
+      return false;
+    }
+
+    // compare size of vectors for measurements and siftIndices
+    if (number_measurements() != sfmTrack.number_measurements() ||
+        siftIndices.size() != sfmTrack.siftIndices.size()) {
+      return false;
+    }
+
+    // compare measurements (order sensitive)
+    for (size_t idx = 0; idx < number_measurements(); ++idx) {
+      SfmMeasurement measurement = measurements[idx];
+      SfmMeasurement otherMeasurement = sfmTrack.measurements[idx];
+
+      if (measurement.first != otherMeasurement.first ||
+          !measurement.second.isApprox(otherMeasurement.second)) {
+        return false;
+      }
+    }
+
+    // compare sift indices (order sensitive)
+    for (size_t idx = 0; idx < siftIndices.size(); ++idx) {
+      SiftIndex index = siftIndices[idx];
+      SiftIndex otherIndex = sfmTrack.siftIndices[idx];
+
+      if (index.first != otherIndex.first ||
+          index.second != otherIndex.second) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  /// print
+  void print(const std::string& s = "") const {
+    cout << "Track with " << measurements.size() << "measurements\n";
```

Comment:
Maybe not optimal

Reply:
Thanks for the PR, Ayush, looks great.

If this is essentially printing the full track representation, maybe this should be something like
```c++
cout << "Track with " << measurements.size() << "measurements of point" << p.x << "," << p.y << "," << p.z << "\n";
```

Reply:
`p.x` and others wont work. Directly printing out `p`